libsolidity/formal/BMC.cpp

```diff
@@ -694,6 +714,34 @@ pair<vector<smtutil::Expression>, vector<string>> BMC::modelExpressions()
 
 /// Verification targets.
 
+string BMC::targetDescription(BMCVerificationTarget const& target)
+{
+	if (
+		target.type == VerificationTargetType::Underflow ||
+		target.type == VerificationTargetType::Overflow
+	)
+	{
+		auto const* intType = dynamic_cast<IntegerType const*>(target.expression->annotation().type);
+		if (!intType)
+			intType = TypeProvider::uint256();
+
+		if (target.type == VerificationTargetType::Underflow)
+			return "Underflow (resulting value less than " + formatNumberReadable(intType->minValue()) + ")";
+		else if (target.type == VerificationTargetType::Overflow)
+			return "Overflow (resulting value larger than " + formatNumberReadable(intType->maxValue()) + ")";
+		else
+			solAssert(false);
```

Comment:
Is this necessary? The `if` block condition should make this unreachable.

Reply:
That's why the asserts are added, no? Precisely to cover impossible situations (or those highly unlikely). What I don't like here is the `else -> assert` instead of just `assert`, since the else here is redundant.

Reply:
Yes, I even thought about writing something like that in my comment, but this seems an overkill, no? I mean, it is just inside the `if` block that checks for those two possibilities...but, either way, it is not really a problem.

Reply:
yea true, the other ifs are enough, will remove this branch